crates/python/qcs_sdk/qpu/api.pyi

```diff
@@ -159,10 +167,45 @@ async def retrieve_results_async(
     :param quantum_processor_id: The ID of the quantum processor the job ran on.
     :param client: The ``QCSClient`` to use. Creates one using environment configuration if unset - see https://docs.rigetti.com/qcs/references/qcs-client-configuration
     :param endpoint_id: retrieve the results of a program submitted to an explicitly provided endpoint. If `None`, the default endpoint for the given quantum_processor_id is used.
+    :param connection_strategy: The ``ConnectionStrategy`` to use to connect to the QPU. If unset, the default is `ConnectionStrategy.GatewayOnly`. If `enpodint_id` is set, then direct access to that endpoint overrides this parameter.
 
     :returns: results from execution.
 
     :raises LoadClientError: If there is an issue loading the QCS Client configuration.
     :raises SubmissionError: If there was a problem during program execution.
     """
     ...
+
+@final
+class ExecutionOptions:
+    @staticmethod
+    def default() -> ExecutionOptions:
+        """Return ExecutionOptions with a reasonable set of defaults."""
+        ...
+
+@final
+class ExecutionOptionsBuilder:
+    def __new__(cls) -> ExecutionOptionsBuilder: ...
+    @staticmethod
+    def default() -> ExecutionOptionsBuilder:
+        """Return a builder with the default values for ``ExecutionOptions``"""
+        ...
+    def connection_strategy(
+        self, connection_strategy: ConnectionStrategy
+    ) -> ExecutionOptionsBuilder:
+        """Set the ``ConnectionStrategy`` used to establish a connection to the QPU."""
+    def build(self) -> ExecutionOptions:
+        """Build the ``ExecutionOptions`` using the options set in this builder."""
+
+@final
+class ConnectionStrategy(Enum):
+    """An enum containing variants for each possible quantum processor connection strategy."""
+
+    GatewayOnly = "GatewayOnly"
+    """
+    Always connect through the publicly accessible gateway.
+    """
+    DirectAccess = "DirectAccess"
```

Comment:
```suggestion
    Gateway = "Gateway"
    """
    Always connect through the publicly accessible gateway.
    """
    DirectAccess = "DirectAccess"
```

To extend @Shadow53 's point, seems like these would benefit from consistency - either both `Only` or neither